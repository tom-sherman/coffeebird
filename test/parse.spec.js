/* global it, expect */
const { CoffeebirdParser } = require('../src/coffeebird')
const { loadExample } = require('./util')

it('should parse comments at the end of the line', () => {
  const comments = loadExample('comments.coffeebird')
  const parse = () => CoffeebirdParser.tryParse(comments)
  expect(parse).not.toThrow()
  expect(parse()).toMatchSnapshot()
})
