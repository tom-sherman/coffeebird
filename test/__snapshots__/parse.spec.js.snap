// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse comments at the end of the line 1`] = `
Array [
  Object {
    "end": Object {
      "column": 42,
      "line": 1,
      "offset": 41,
    },
    "name": "comment",
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
    "value": " Comments have to sit on their own line",
  },
  Object {
    "end": Object {
      "column": 15,
      "line": 2,
      "offset": 56,
    },
    "name": "concept",
    "start": Object {
      "column": 1,
      "line": 2,
      "offset": 42,
    },
    "value": Object {
      "name": "Person",
      "options": Object {},
    },
  },
  Object {
    "end": Object {
      "column": 37,
      "line": 4,
      "offset": 94,
    },
    "name": "comment",
    "start": Object {
      "column": 1,
      "line": 4,
      "offset": 58,
    },
    "value": " Block comments are not supported.",
  },
  Object {
    "end": Object {
      "column": 63,
      "line": 5,
      "offset": 157,
    },
    "name": "comment",
    "start": Object {
      "column": 1,
      "line": 5,
      "offset": 95,
    },
    "value": " But you can always add multiple comments on separate lines.",
  },
]
`;
